kernel/shared/src/main/scala/cats/effect/kernel/Async.scala

```diff
@@ -166,6 +166,20 @@ trait Async[F[_]] extends AsyncPlatform[F] with Sync[F] with Temporal[F] {
    */
   def evalOn[A](fa: F[A], ec: ExecutionContext): F[A]
 
+  /**
+   * [[Async.evalOn]] with provided java.util.concurrent.Executor
+   */
+  def evalOn[A](fa: F[A], executor: Executor): F[A] =
+    executor match {
+      case ec: ExecutionContextExecutor =>
```

Comment:
It's possible it implements `ExecutionContext` without implementing `ExecutionContextExecutor`.
```suggestion
      case ec: ExecutionContext =>
```